telnyx_rtc/src/main/java/com/telnyx/webrtc/sdk/TelnyxClient.kt

```diff
@@ -1929,6 +1932,9 @@ class TelnyxClient(
         invalidateGatewayResponseTimer()
         resetGatewayCounters()
         unregisterNetworkCallback()
-        socket.destroy()
+
+        Handler(Looper.getMainLooper()).postDelayed(Runnable {
```

Comment:
I'm not sure how I feel about this. 

we onDisconnect manually sometimes to destroy the socket to ensure we get push notifications. I don't know about adding an artificial delay. 

On the other hand, 500ms isn't huge in the grand scheme of things. I would prefer to try solve this on the client side somehow though...